server/serverAPI/controller/UserController.ts

```diff
@@ -0,0 +1,169 @@
+import { Request, Response } from "express";
+import IDatabase from '../../database/IDatabase';
+import ResponseFormatter from "../../utils/ResponseFormatter";
+import { ResponseTypes } from "../../utils/ResponseTypes";
+import IFood from "../model/food/IFood";
+import IBaseInstruction from "../model/instruction/IBaseInstruction";
+import IInstruction from "../model/instruction/IInstruction";
+import IUser from "../model/user/IUser";
+
+/**
+ * This class creates several properties responsible for user-actions 
+ * provided to the user.
+ */
+export default class UserController {
+    private database: IDatabase<IUser>;
+
+    constructor(database: IDatabase<IUser>) {
+        this.database = database;
+    }
+
+    private getException(error: unknown): string {
+        if (error instanceof Error) {
+            return error.message;
+        }
+
+        return String(error);
+    }
+
+    /**
+     * Lets client to get information about all users existed on the server.
+     * Upon successful operation, this handler will return all users (including their non-sensitive information) existed on the server. 
+     * 
+     * @param req Request parameter that holds information about request
+     * @param res Response parameter that holds information about response
+     */
+    getUsers = async (req: Request, res: Response) => {
```

Comment:
There is no way to make it "private". I use it personally for testing purposes to see what kind of information is leaking and whatnot. Personally, I don't think having this endpoint is bad even if frontend doesn't need it simply because I use it.